### PR TITLE
fix for 522

### PR DESCRIPTION
**Pull request by dougkenik**
Fix for #522

### CODE REVIEW

File: SmartSlicePlugin/stage/SmartSliceScene.py [hunk 1]
```diff
@@ -529,13 +529,36 @@ def setArrow(self, direction: Vector):
...
+
+            # Project the vector into the phi plane
+            phi_vector = direction - direction.dot(self.phiRotator.rotation_axis.normalized()) * self.phiRotator.rotation_axis.normalized()
+
+            angle = angleBetweenVectors(self.activeArrow.direction, phi_vector)
+
+            axes_angle = angleBetweenVectors(self.phiRotator.rotation_axis, phi_vector.cross(self.activeArrow.direction))
+            angle = -angle if abs(axes_angle) < 1.e-3 else angle
+
+            self.rotateArrow(angle, RotationAxis.Phi)
```
Comment (bradymadams) [inline SmartSlicePlugin/stage/SmartSliceScene.py:560]: Lines 550-560 are repeated a few times. Wouldn't mind seeing those eventually consolidated.

**Review by bradymadams — APPROVED** [review]
It is behaving as expected now. The load arrows remain after clicking off the part, but that's because the fixes in #527 aren't in this PR.

**Comment by rickdalgarno** [comment]
@dougkenik The load wheels still have some issues:

1. Define custom load and move both wheels.
2. During validation, adjust the angle slider, and click "Cancel" when prompted.

After clicking Cancel, the wheels re-orient in a seemingly random fashion. The load arrow, however, does appear to maintain the same direction.

I tested using Cura 4.7 and 4.9.

**Comment by dougkenik** [comment]
> @dougkenik The load wheels still have some issues:
> 
> 1. Define custom load and move both wheels.
> 2. During validation, adjust the angle slider, and click "Cancel" when prompted.
> 
> After clicking Cancel, the wheels re-orient in a seemingly random fashion. The load arrow, however, does appear to maintain the same direction.
> 
> I tested using Cura 4.7 and 4.9.
> 
> 

@rickdalgarno - if the arrow returns to the original direction, then I would say we are good. We do not record the steps the user took to get to that orientation, so it is near impossible to put the rotators in the same position.

**Review by rickdalgarno — APPROVED** [review]
Tested using Cura 4.7 and 4.9 and build SmartSlicePlugin-r21.0-522.29. This fixes the issues in #522.

**Pull request merged**